File: kotlinames-common/src/main/java/com/nagopy/android/kotlinames/property/required/KRequiredDatePropertyName.java
package com.nagopy.android.kotlinames.property.required;

import com.nagopy.android.kotlinames.KRequiredPropertyName;
import com.nagopy.android.kotlinames.KSortablePropertyName;

import java.util.Date;

public class KRequiredDatePropertyName implements KRequiredPropertyName<Date>, KSortablePropertyName<Date> {

    private final String name;

    public KRequiredDatePropertyName(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }
}

File: kotlinames/src/main/kotlin/com/nagopy/android/kotlinames/property/required/KRequiredDatePropertyName.kt
package com.nagopy.android.kotlinames.property.required

import com.nagopy.android.kotlinames.KRequiredPropertyName
import com.nagopy.android.kotlinames.KSortablePropertyName
import java.util.*

class KRequiredDatePropertyName(val name: String) : KRequiredPropertyName<Date>, KSortablePropertyName<Date> {

    override fun name(): String = name

}
